File: grpc-examples/go/app/server/other.go
package server

import (
	"context"
	"log"
	"net"
	"time"

	"google.golang.org/grpc"

	pb "github.com/t10471/go-examples/grpc-examples/go/proto"
)

func OtherMain(myAddress string, sleep int) error {
	lis, err := net.Listen("tcp", myAddress)
	if err != nil {
		log.Printf("failed to listen: %v", err)
		return err
	}
	s := grpc.NewServer()
	pb.RegisterOtherServer(s, &other{sleep: time.Duration(sleep) * time.Second})
	log.Printf("server listening at %v", lis.Addr())
	if err := s.Serve(lis); err != nil {
		log.Printf("failed to serve: %v", err)
		return err
	}
	return nil
}

type other struct {
	pb.UnimplementedOtherServer
	sleep time.Duration
}

func (s *other) CallOther(ctx context.Context, in *pb.OtherRequest) (*pb.OtherReply, error) {
	log.Printf("Received: %v", in.GetName())
	log.Printf("sleep: %v", s.sleep)
	time.Sleep(s.sleep)
	log.Printf("sleep end")
	return &pb.OtherReply{Message: "Hello " + in.GetName()}, nil
}

func (s *other) CallOtherV2(ctx context.Context, in *pb.OtherRequest) (*pb.OtherReply, error) {
	log.Printf("Received: %v", in.GetName())
	return &pb.OtherReply{Message: "Hello " + in.GetName()}, nil
}

File: grpc-examples/python/app/server/other.py
import asyncio
import logging

import grpc
import proto.other_pb2 as other_pb2
import proto.other_pb2_grpc as other_pb2_grpc


class Other:
    def main(self, my_address: str, sleep_sec: int):
        asyncio.run(serve(my_address, sleep_sec))


class OtherCallee(other_pb2_grpc.OtherServicer):
    def __init__(self, sleep_sec: int):
        self.sleep_sec = sleep_sec

    async def CallOther(
        self, request: other_pb2.OtherRequest, context: grpc.aio.ServicerContext
    ) -> other_pb2.OtherReply:
        logging.info("start sleep")
        await asyncio.sleep(self.sleep_sec)
        logging.info("end sleep")
        return other_pb2.OtherReply(message="Hello, %s!" % request.name)

    async def CallOtherV2(
        self, request: other_pb2.OtherRequest, context: grpc.aio.ServicerContext
    ) -> other_pb2.OtherReply:
        return other_pb2.OtherReply(message="Hello, %s!" % request.name)


async def serve(my_address: str, sleep_sec: int) -> None:
    server = grpc.aio.server()
    other_pb2_grpc.add_OtherServicer_to_server(OtherCallee(sleep_sec), server)
    listen_addr = my_address
    server.add_insecure_port(listen_addr)
    logging.info("Starting server on %s", listen_addr)
    await server.start()
    await server.wait_for_termination()
